media/CfgEditor/cfgeditor.js

```diff
@@ -13,32 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/*
- * Copyright (c) Microsoft Corporation
- *
- * All rights reserved.
- *
- * MIT License
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
- * associated documentation files (the "Software"), to deal in the Software without restriction,
- * including without limitation the rights to use, copy, modify, merge, publish, distribute,
- * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in all copies or
- * substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
- * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
- * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
- */
-/*
-This code refers to
-https://github.com/microsoft/vscode-webview-ui-toolkit-samples/blob/b807107df40271e83ea6d36828357fdb10d71f12/default/hello-world/webview-ui/main.js
-*/
+
```

Comment:
Q. How about transforming to `ts`?

Reply:
Is it possible? or is it making you feel difficult? (if so, I would do it after landing this code.)

Reply:
At first, I am saying that I can be wrong because I am not good at HTML and web.
So if there are something wrong, please let me know :)

---

AKAIK, HTML only understands javascript, not typescript.
Therefore, if we want to use typescript in HTML, we should compile `.ts` file first and then include the compiled `.js` file.
However, current structure for our repository seems not allow to do so :(
Briefly speaking, current structure is as following.

- `src` directory : includes **dynamic** `ts` files for using VSCode extension APIs
- `media` directory : includes **static** `html`, `css`, `js` files which are loaded on webview

With this status, let's say that I created `cfgeditor.ts` instead of `cfgeditor.js`.
Then how should I handle this?

- If I put `cfgeditor.ts` to `src` directory, it is not dynamic file.
- If I put `cfgeditor.ts` to `static` directory, the file is not compiled so we cannot use it directly.

In conclusion, I think that's why MS introduced sample VSCode extension as current structure.
(https://github.com/microsoft/vscode-extension-samples/tree/main/webview-view-sample)

If you have a good idea to transforming these `.js` files to `.ts` files, let's discuss about it together :D

Reply:
Now I read your kind comment. Thanks!